src/ItspServices.pServer.Persistence.Sqlite/Repositories/UserRepository.cs

```diff
@@ -15,125 +15,97 @@ class UserRepository : IUserRepository
                 {
-                    query.AddParameterWithValue("searchedUsername", name);
-                    query.CommandText = "SELECT * FROM [Users Keys] WHERE Username=@searchedUsername;";
-                    using (IDataReader reader = query.ExecuteReader())
+                    int i = 0;
+                    insert.CommandText = "INSERT OR IGNORE INTO Roles(name) VALUES";
+                    foreach (string role in serverRoles)
                     {
-                        if (reader.Read())
-                        {
-                            user.Id = reader.GetInt32(0);
-                            user.UserName = reader.GetString(1);
-                            user.NormalizedUserName = reader.GetString(2);
-                            user.PasswordHash = reader.GetString(3);
-                            user.Role = reader.GetString(4);
-                            if (!reader.IsDBNull(5))
-                            {
-                                user.PublicKeys.Add(new Key()
-                                {
-                                    Id = reader.GetInt32(5),
-                                    KeyData = Convert.FromBase64String(reader.GetString(6)),
-                                    Flag = reader.GetBoolean(7) ? Key.KeyFlag.ACTIVE : Key.KeyFlag.OBSOLET
-                                });
-                            }
-                        }
-                        while(reader.Read())
-                        {
-                            user.PublicKeys.Add(new Key()
-                            {
-                                Id = reader.GetInt32(5),
-                                KeyData = Convert.FromBase64String(reader.GetString(6)),
-                                Flag = reader.GetBoolean(7) ? Key.KeyFlag.ACTIVE : Key.KeyFlag.OBSOLET
-                            });
-                        }
+                        insert.AddParameterWithValue($"role{i}", role);
+                        insert.CommandText += (i < serverRoles.Count - 1) ? $"(@role{i})," : $"(@role{i})";
+                        i++;
                     }
+                    insert.ExecuteNonQuery();
                 }
             }
-            return user;
         }
 
-        public User GetById(int id)
+        public User GetUserByNormalizedName(string name)
         {
-            User user = ReadUserData(id);
-            if(user == null)
-            {
-                return null;
-            }
-            else
+            User user;
+            using (DbConnection con = _dbFactory.CreateAndOpenConnection(_connectionString))
             {
-                AddPublicKeys(user);
-                return user;
+                using (DbCommand query = con.CreateCommand())
+                {
+                    query.AddParameterWithValue("searchedUsername", name);
+                    query.CommandText = "SELECT * FROM [Users Keys] WHERE NormalizedUsername=@searchedUsername;";
```

Comment:
I wouldn't use * here, but all the fields that are required. It makes it clearer, and prevents loading fields that might be  there and expensive in the future and are not required.